vllm/forward_context.py

```diff
@@ -94,7 +94,11 @@ class ForwardContext:
     virtual_engine: int  # set dynamically for each forward pass
     # set dynamically for each forward pass
     dp_metadata: Optional[DPMetadata] = None
-    skip_cuda_graphs: bool = False
+    # determine whether to use a full cudagraph for attention or piecewise
+    # cudagraphs that skip the attention part. By default true, we use piecewise
+    # cudagraphs.
+    skip_attention_cuda_graphs: bool = True
+    is_pure_decoding: bool = False
```

Comment:
I would name this `is_pure_decode` or `is_pure_decode_batch` (but matter of preference)